conans/client/build/autotools_environment.py

```diff
@@ -339,6 +341,11 @@ def append(*args):
         tmp_compilation_flags = copy.copy(self.flags)
         if self.fpic:
             tmp_compilation_flags.append(pic_flag(self._conanfile.settings))
+        if tools.is_apple_os(self._os):
+            concat = " ".join(tmp_compilation_flags)
+            if self._os_version and "-version-min" not in concat and "-target" not in concat:
+                tmp_compilation_flags.append(tools.apple_deployment_target_flag(self._os,
```

Comment:
I would like to understand a bit better (not a OSX user) what is the effect of not adding this flag. Is it a compilation error? What have been the users doing so far for this?

Also, is there a chance that this could be breaking for other users that do not expect this flag to be added? Should this flag be added in other generators as well (cmake, xcode)?

Reply:
If [this flag](https://clang.llvm.org/docs/CommandGuide/clang.html#target-selection-options) is not set and someone is trying to link a library with no minimal os version against an application explicitely specifying it then warning will span like mentioned here: https://github.com/bincrafters/community/issues/1264#issuecomment-705448208

Regarding the users that do not expect this flag to be added I wondered why they would have specify the `os.version` setting that (if I'm understanding well) states that the dependencies should support such OS.

Cmake generators allready handle the `os.version` flags by setting the CMAKE_OSX_DEPLOYMENT_TARGET definition: https://github.com/conan-io/conan/blob/develop/conans/client/build/cmake_flags.py#L216-L219